astropy/time/core.py

```diff
@@ -1318,13 +1359,13 @@ def __add__(self, other):
         # ideally, we calculate in the scale of the Time item, since that is
         # what we want the output in, but this may not be possible, since
         # TimeDelta cannot be converted arbitrarily
+        out = self.replicate()
         if self.scale in other.SCALES:
-            out = self.replicate()
             if other.scale not in (out.scale, None):
                 other = getattr(other, out.scale)
         else:
-            out = getattr(self, (other.scale if other.scale is not None
-                                 else 'tai'))
+            out = getattr(out.replicate(), (other.scale if other.scale is not None
```

Comment:
The `replicate` here is not needed any more, since it is done above (a leftover?)